core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -2334,10 +2334,14 @@ object IO extends IOCompanionPlatform with IOLowPriorityImplicits with TuplePara
 }
 
 private object SyncStep {
-  def interpret[G[+_], B](io: IO[B], limit: Int)(
+  final val MaxSteps = 512 // 512 matches  with trampoline depth in IOFiber for stack safety
+
+  def interpret[G[+_], B](io: IO[B], limit: Int, stepsUntilDefer: Int)(
```

Comment:
Another optimization, so that it compiles as a static method.

```suggestion
  @static def interpret[G[+_], B](io: IO[B], limit: Int, stepsUntilDefer: Int)(
```

Reply:
adding static is failing compilation as `SyncStep` is private object and `@static` requires a companion class